app/uk/gov/hmrc/cdsreimbursementclaim/models/tpi01/ClaimsResponse.scala

```diff
@@ -27,18 +28,27 @@ object ClaimsResponse {
 
   def fromTpi01Response(responseDetail: ResponseDetail): ClaimsResponse = {
 
-    val scty = responseDetail.CDFPayCase
+    val scty       = responseDetail.CDFPayCase
       .flatMap(_.SCTYCases)
       .getOrElse(Seq.empty)
       .map(SctyClaimItem.fromTpi01Response)
       .filter(_.declarationID.isDefined)
+    val uniqueScty = removeDuplicates[SctyClaimItem](scty, _.CDFPayCaseNumber)
 
-    val ndrc = responseDetail.CDFPayCase
+    val ndrc       = responseDetail.CDFPayCase
       .flatMap(_.NDRCCases)
       .getOrElse(Seq.empty)
       .map(NdrcClaimItem.fromTpi01Response)
       .filter(_.declarationID.isDefined)
+    val uniqueNdrc = removeDuplicates[NdrcClaimItem](ndrc, _.CDFPayCaseNumber)
 
-    ClaimsResponse(scty, ndrc)
+    ClaimsResponse(uniqueScty, uniqueNdrc)
   }
+  def removeDuplicates[A](list: Seq[A], get: A => String): Seq[A] =
+    list.foldLeft(List.empty[A]) { (acc, elem) =>
+      acc.find(item => get(item) === get(elem)) match {
+        case Some(_) => acc
+        case None    => acc :+ elem
```

Comment:
This will get the worst possible performance, probably better to reverse the List after folding

Reply:
😓 That might be better since List in scala is actually a linked list